### PR TITLE
Handling Exceptions

### LINKED ISSUE #118: Exception handling for Phoneme Factory
**Describe the solution you'd like**

The phoneme factory should be able to handle the following exceptions from the user inputs/Rhubarb Process:

1) FileNotFound error for audio and text filepath inputs
2) Ensuring that files are the correct type (audio must be .wav, text must be .txt) 
3) Ensuring that file contents are not corrupt (.wav must be valid and .txt must be valiud)
4) Checking that files contain information and are not empty or very short 
5) Checking that the output.json exists, along with mouthCues array 

**Additional context**

There are two methods we could perform exception handling:

1) Outside the Child process 
- We perform exception handling before inputs are tossed into Child process
- Validate the file paths and files against the existing files in the directory before executing the Child process 

2) Inside/after the Child process 
- We catch errors and have event listeners for the different errors that may come about 

This might be helpful: 
https://stackoverflow.com/questions/34208614/how-to-catch-an-enoent-with-nodejs-child-process-spawn

### PR DESCRIPTION
**Pull request by vnajack**
- closes #118 

Currently, the file checks are running, but the Rhubarb process starts running before the checks are complete. Please help me brainstorm ideas on how to handle this!

Ed and I were looking at using promises, async, await, but the functions in the `file_check.ts` will either throw an error or print a console command. This results in a few difficulties trying to make sure things run in a specific order.

### CODE REVIEW

File: docs/reports/peer_testing_1/usability_tasks.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+and will be adding more customizable assets for the avatars in the future.
+
+Since our product is not yet wrapped into an API, nor are any frames being produced, we don't have any developer scenarios yet.
+
+Stay tuned for developer scenarios in Peer Testing II.
+
+### Developer Scenario 1: Change the Mouth Assets
+
+**(This might not be included depending on our progress this week)
+Change out the mouth assets (files) for new ones.
```
Comment (LoganParker) [inline docs/reports/peer_testing_1/usability_tasks.md:38]: not sure if we currently have a mouth asset directory, but some of the work on branch ffmpef-exploration includes these assets

File: speaking-portal-project/src/phonemeFactory/rhubarb.ts [hunk 2]
```diff
@@ -1,25 +1,36 @@
...
+
+    const args = [
+        "-o ","output.json",
+        "--exportFormat","json",
+        "-r ",`${recognizer}`,
+        "-d",`${text_file_name}`,
+        "--extendedShapes", "GX", `${audio_file_name}`
+    ]
+
+    const rhubarbProc = spawnSync('./rhubarb', args)
```
Comment (LoganParker) [inline speaking-portal-project/src/phonemeFactory/rhubarb.ts:32]: Nice work getting spawn implemented

File: speaking-portal-project/src/test/file_check.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    checkFileExistence(audio_file_name)
+    checkFileReadAccess(audio_file_name)
+    checkWavFile(audio_file_name)
+
+    // Check Text File
+    checkFileExistence(text_file_name)
+    checkFileReadAccess(text_file_name)
+    checkTextFile(text_file_name)
+
+    // TODO: chdir() back to original directory? or do we do this after running Rhubarb? should it be here, in index.ts or in rhubarb.ts?
```
Comment (LoganParker) [inline speaking-portal-project/src/test/file_check.ts:25]: Each time we change directory to the rhubarb directory, we should be switching back to the main directory

**Review by LoganParker — APPROVED** [review]
These changes look great!! I've rebased the ffmpeg exploration branch to this one to resolve any potential merge conflicts

File: docs/reports/peer_testing_1/usability_tasks.md (hunk 1 above)
Comment (vnajack) [reply docs/reports/peer_testing_1/usability_tasks.md:38]: we will want to give the option for them to switch the mouths to something else, but I'm not sure if this is a task we want for Peer Testing I or Peer Testing II

File: speaking-portal-project/src/phonemeFactory/rhubarb.ts (hunk 2 above)
Comment (vnajack) [reply speaking-portal-project/src/phonemeFactory/rhubarb.ts:32]: this was all @edouarde1 !!

**Review by edouarde1 — APPROVED** [review]
Great work! I am using these exceptions at the moment for unit testing.

We made need to make some modifications down the line, we can look into that this weekend and during the week!

**Pull request merged**